pkgs/development/compilers/unison/default.nix

```diff
@@ -4,18 +4,18 @@
 
 stdenv.mkDerivation rec {
   pname = "unison-code-manager";
-  milestone_id = "M1h";
+  milestone_id = "M1j";
   version = "1.0.${milestone_id}-alpha";
 
   src = if (stdenv.isDarwin) then
     fetchurl {
       url = "https://github.com/unisonweb/unison/releases/download/release/${milestone_id}/unison-osx.tar.gz";
-      sha256 = "0iivm5gmbk0fq0zr3lvck6p1c2i7i54l3rf70z677529w9irzchp";
+      sha256 = "1pvdjmasgl22inbr8nlizsg8s5zagn8bzwhaxqmwafkpsskz0hsg";
     }
   else
     fetchurl {
       url = "https://github.com/unisonweb/unison/releases/download/release/${milestone_id}/unison-linux64.tar.gz";
-      sha256 = "0fb84c1yn8pidflh7kq696j3v4blkvbk1fsqp36h30p7vv676yci";
+      sha256 = "0ybs7y7p4wlhzl3vsn60wx9icf8pv09g2r6qcr2pyar76vd142sw";
```

Comment:
This hash should be `1xpblx405cp3mv0vrhcqwjlxvrhgmc77mxbvcy93srxja3qai1af`

Reply:
```bash
$ nix-prefetch-url  https://github.com/unisonweb/unison/releases/download/release/M1j/unison-linux64.tar.gz
[7.2 MiB DL]
path is '/nix/store/d6r6vvd6w1lamg37hh9dqqg4gi3mls8n-unison-linux64.tar.gz'
1xpblx405cp3mv0vrhcqwjlxvrhgmc77mxbvcy93srxja3qai1af
```

Reply:
Oh. I had done `nix-prefetch-url --unpack https://github.com/unisonweb/unison/releases/download/release/M1j/unison-linux64.tar.gz`, which resulted in a different hash. When should you use `--unpack` and when shouldn't you? I tried finding more information on this and couldn't find much. I think that I did the same for the Darwin file, so maybe that needs to change too? But it worked fine for me when I ran it locally.

Reply:
Oh never mind, looking at my shell history I didn't do the same for Darwin, so it's probably right. Still if someone could shed some light on when to `--unpack` and when not to, I'd appreciate it.

Reply:
@ceedubs `--unpack` is a hash of the content. You will need to use `--unpack` to get hashes for `fetchzip`, since that hashes the content. Not using `--unpack` is a hash of the archive.

Reply:
@worldofpeace ah okay. Thank you for the clarification.

Reply:
@virusdave @worldofpeace 
Sorry to interrupt in, but could/would someone help me understand the nix hashing here better?  When I try to compute it using `shasum` (which is what we use for the ucm homebrew formula), I get a different result from any of the above:
```
$ wget https://github.com/unisonweb/unison/releases/download/release/M1j/unison-linux64.tar.gz
$ shasum -a 256 unison-linux64.tar.gz 
4e85a8f050b2673d92677bf57a0eab0fe6dda9e498c1bcc1aee3b20248a7ebf6  unison-linux64.tar.gz
```
Any ideas why?  Thanks in advance.

Reply:
No, it doesn't seem to match according to any base32 alphabet I can find (rfc4648 base32, z-base-32, crockford's, base32hex, geohash). 😕

Reply:
I guess nix uses [its own alphabet](https://github.com/NixOS/nix/blob/629b9b0049363e091b76b7f60a8357d9f94733cc/src/libutil/hash.cc#L73), though I still can't seem to make the hash bits match up; but maybe it's enough to say that the appropriate hash comes from the nix tools (and not from elsewhere) and that's okay :)

Reply:
> @virusdave @worldofpeace
> Sorry to butt in, but could/would someone help me understand the nix hashing here better? When I try to compute it using `shasum` (which is what we use for the ucm homebrew formula), I get a different result from any of the above:
> 
> ```
> $ wget https://github.com/unisonweb/unison/releases/download/release/M1j/unison-linux64.tar.gz
> $ shasum -a 256 unison-linux64.tar.gz 
> 4e85a8f050b2673d92677bf57a0eab0fe6dda9e498c1bcc1aee3b20248a7ebf6  unison-linux64.tar.gz

```
$ nix-hash --to-base16 --type sha256 1xpblx405cp3mv0vrhcqwjlxvrhgmc77mxbvcy93srxja3qai1af

4e85a8f050b2673d92677bf57a0eab0fe6dda9e498c1bcc1aee3b20248a7ebf6
```

They match :)